std/experimental/logger/core.d

```diff
@@ -3105,3 +3104,43 @@ private void trustedStore(T)(ref shared T dst, ref T src) @trusted
     tl.logf("%s", sts);
     assert(tl.msg == SystemToStringMsg);
 }
+
+// Issue 17328
+@safe unittest
+{
+    import std.format : format;
+
+    ubyte[] data = [0];
+    string s = format("%(%02x%)", data); // format 00
+    assert(s == "00");
+
+    auto tl = new TestLogger();
+
+    tl.infof("%(%02x%)", data);    // infof    000
+
+    size_t i;
+    string fs = tl.msg;
+    for (; i < s.length; ++i)
+    {
+        assert(s[s.length - 1 - i] == fs[fs.length - 1 - i], fs);
+    }
```

Comment:
because s and fs only have the postfix, otherwise they are different strings, and I need to make sure that the last two chars are equal